### DISCUSSION TITLE
Are booleans case sensitive in fail2ban config files?

### DISCUSSION THREAD

**Discussion by chriscroome** [fail2ban/fail2ban#3886] [Q&A] [answered]
I'm looking at using the Ansible [community.general.to_ini_filter](https://docs.ansible.com/ansible/latest/collections/community/general/to_ini_filter.html) to template fail2ban config files and if I do it will result in boolean values that are currently lower case being capitalized, is this fine or do I need to ensure they are lower case?

For example in `/etc/fail2ban/jail.local`:

```ini
[DEFAULT]
ignoreip = 127.0.0.1
bantime = 360000
banaction = iptables-multiport
logpath = /var/log/auth.log
[sshd]
enabled = True  # Currently I have a lower case true
bantime = 86400
```

**Comment by sebres** [comment, marked as answer]
By conversion from string (what basically happens in config), it is case-**insensitive** (due to `lower()` in `_as_bool` helper), moreover it can be additionally `1`, `on`, or `yes` (both latter also case-insensitive):
https://github.com/fail2ban/fail2ban/blob/12ff98027f82d4d6d02a9cc8fb0b23f1a817a34f/fail2ban/helpers.py#L88-L90

**Reply by chriscroome** [reply]
Thanks!
